### PR TITLE
unixPb: Add role to install intel homebrew for arm64 macs

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/2536#issuecomment-1708716478

~Need to test before merging~

### CODE REVIEW

**Comment by Haroon-Khel** [comment]
Tested. Works as expected. Ready to merge

**Comment by Haroon-Khel** [comment]
Going forward, any brew install ansible task which intends to run on arm64 mac needs to have its path specified to /opt/homebrew/bin/brew seeing as we will now have both intel and arm64 homebrew on our machines

**Comment by Haroon-Khel** [comment]
If this gets merged now AWX will run the changes on our build machines, so i recommend it gets merged until after the coming release

**Comment by karianna** [comment]
Why do we need Intel brew on Arm?

**Comment by Haroon-Khel** [comment]
@karianna arm64 macs need the intel library `libpng` to be able to compile freetype for x64 jdk8 which is installed using intel homebrew

See https://github.com/adoptium/infrastructure/issues/2536#issuecomment-1708716478

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml [hunk 1]
```diff
@@ -25,12 +25,12 @@
 
 - name: Set homebrew path (x64)
   set_fact:
-    homebrew_path: /usr/local/bin/brew
+    homebrew_path: /usr/local/bin
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml:28]: will this leave old installs lying about?

**Review by karianna — APPROVED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml (hunk 1 above)
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml:28]: @karianna I believe we're ok. `homebrew_path` is being changed from the executable location to the directory that the executable was in. The only two previous references to this variable were the check for its presence and one invocation, which have been modified. All the others are new references and I believe the `/usr/local/bin` and `/opt/homebrew/bin` locations are the defaults so this should not conflict with existing installation. I'll be honest the use of the `/bin` in these paths looks a bit strange to me, but that is consistent with the [ansible docs](https://docs.ansible.com/ansible/latest/collections/community/general/homebrew_module.html#parameter-path).

**Comment by karianna** [comment]
OK, so just checks to go green and this is good

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Homebrew_intel/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  stat:
+    path: /usr/local/Homebrew/bin/brew
+  register: intel_homebrew_installed
+
+- name: Run Homebrew install script in a Rosetta shell
+  ansible.builtin.shell: yes | arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  when: not intel_homebrew_installed.stat.exists
+
+- name: Install Intel libpng
+  ansible.builtin.shell: arch -x86_64 yes | /usr/local/Homebrew/bin/brew install libpng
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Homebrew_intel/tasks/main.yml:16]: This is presumably the bit that caused the problems from https://github.com/adoptium/infrastructure/issues/3231 so I'm going to block this change on the basis that we presumably fix this PR before it goes live.

**Review by sxa — CHANGES_REQUESTED** [review]

**Comment by Haroon-Khel** [comment]
As per https://github.com/adoptium/temurin-build/issues/3509#issuecomment-1790995038, we do not need libpng installed (either arm64 or intel) on our build machines to be able to cross compile x64 mac jdk8 which means we do not need intel homebrew on our arm64 macs

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml [hunk 3]
```diff
@@ -162,6 +180,14 @@
   when: not pdfwriter.stat.exists
   tags: jck_tools
 
+- name: Uninstall libpng
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml:183]: I'm `-1` on this as I don't want end-users who use our playbooks to have things removed from their machine. Suggest we do this as a one off on all of our machines unless there is a specific reason for this to be in the playbooks. If it has to be, let's tag it with `adoptopenjdk`.

Also bear in mind that [we have a bug](https://github.com/adoptium/temurin-build/issues/3509) which is causing libpng to be picked up even when we tell it not to, which is not ideal :-)

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml:183]: (Now tagged with adoptopenjdk so objection removed but we should plan to remove this in the future if we can)

**Review by sxa — APPROVED** [review]
OK I think this is good to go in now :-) Thanks for the changes.

**Pull request merged**